.ci/install_kata.sh

```diff
@@ -10,7 +10,7 @@ set -o nounset
 set -o pipefail
 
 cidir=$(dirname "$0")
-tag="${1:-""}"
+tag="${1:-"vanilla"}"
```

Comment:
I don't remember where I mentioned that originally this `tag` option was meant to be a version of kata (i.e. a tag in the git repository). Here we are changing the semantic of `tag` completely.

What if we do the following:
 - Introduce two new variables: `KATA_BUILD_KERNEL_TYPE`  and `KATA_BUILD_QEMU_TYPE` (don't need to be exactly those names). They can have the values of `vanilla` (default if not set), `experimental` and `tdx`.
 - In `install_kata.sh` have a block to initialize those (code below is just a draft):
   ```
   KATA_BUILD_KERNEL_TYPE="${KATA_BUILD_KERNEL_TYPE:-vanilla}"
   KATA_BUILD_QEMU_TYPE="${KATA_BUILD_QEMU_TYPE:-vanilla}"
   if [ "${TEE_TYPE:-}" == "tdx" ]; then
       # This allow one to even export a different TDX type, like KATA_BUILD_KERNEL_TYPE=my-tdx-version
       KATA_BUILD_KERNEL_TYPE="${KATA_BUILD_KERNEL_TYPE:-tdx}"
       ....
       ....
   fi
   ```
 - Finally call the respective install scripts with `-t` ( `-t ${KATA_BUILD_KERNEL_TYPE}`, for example).
 - Kill the `experimental_qemu` and `experimental_kernel` variables. Okay, this can be done in a follow up PR as involve many other considerations I believe.

What do you think @arronwy ?
Any opinion on that @devimc, @jcvenegas ?